--- gnu-efi-3.0/lib/x86_64/initplat.c~	2014-08-08 15:08:55.000000000 -0400
+++ gnu-efi-3.0/lib/x86_64/initplat.c	2014-12-10 02:35:47.547768468 -0500
@@ -26,3 +26,27 @@
 {
 }
 
+
+/*
+ * Calls to these functions may be emitted implicitly by GCC even when
+ * -ffreestanding is in effect.
+ */
+void *memset(void *s, int c, __SIZE_TYPE__ n)
+{
+    unsigned char *p = s;
+
+    while (n--)
+        *p++ = c;
+
+    return s;
+}
+
+void *memcpy(void *dest, const void *src, __SIZE_TYPE__ n)
+{
+    unsigned char *p = dest, *q = src;
+
+    while (n--)
+        *p++ = *q++;
+
+    return dest;
+}
