--- gnu-efi-3.0.5/Make.defaults~	2017-01-09 15:12:04.000000000 +0000
+++ gnu-efi-3.0.5/Make.defaults	2017-05-12 17:59:00.138439909 +0000
@@ -157,7 +157,7 @@
            -fshort-wchar -fno-strict-aliasing \
            -ffreestanding -fno-stack-protector
 else
-CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra -Werror \
+CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra \
            -fshort-wchar -fno-strict-aliasing \
            -fno-merge-constants -ffreestanding -fno-stack-protector \
            -fno-stack-check
