--- gnu-efi-3.0.8/lib/init.c~	2018-03-14 18:54:51.000000000 +0000
+++ gnu-efi-3.0.8/lib/init.c	2018-05-05 17:37:00.319989272 +0000
@@ -192,7 +192,7 @@
 #define __SIZE_TYPE__ UINTN
 #endif
 
-void *memset(void *s, int c, __SIZE_TYPE__ n)
+ __attribute__ ((weak)) void *memset(void *s, int c, __SIZE_TYPE__ n)
 {
     unsigned char *p = s;
 
@@ -202,7 +202,7 @@
     return s;
 }
 
-void *memcpy(void *dest, const void *src, __SIZE_TYPE__ n)
+ __attribute__ ((weak)) void *memcpy(void *dest, const void *src, __SIZE_TYPE__ n)
 {
     const unsigned char *q = src;
     unsigned char *p = dest;
